addons/procurement/procurement.py

```diff
@@ -60,177 +52,143 @@ class procurement_rule(osv.osv):
+    def run(self, autocommit=False):
+        for procurement in self:
             if procurement.state not in ("running", "done"):
                 try:
-                    if self._assign(cr, uid, procurement, context=context):
-                        res = self._run(cr, uid, procurement, context=context or {})
+                    if self._assign(procurement):
+                        res = self._run(procurement)
                         if res:
-                            self.write(cr, uid, [procurement.id], {'state': 'running'}, context=context)
+                            procurement.write({'state': 'running'})
                         else:
-                            self.write(cr, uid, [procurement.id], {'state': 'exception'}, context=context)
+                            procurement.write({'state': 'exception'})
                     else:
-                        self.message_post(cr, uid, [procurement.id], body=_('No rule matching this procurement'), context=context)
-                        self.write(cr, uid, [procurement.id], {'state': 'exception'}, context=context)
+                        procurement.message_post(body=_('No rule matching this procurement'))
+                        procurement.write({'state': 'exception'})
                     if autocommit:
-                        cr.commit()
+                        self._cr.commit()
                 except OperationalError:
                     if autocommit:
-                        cr.rollback()
+                        self._cr.rollback()
                         continue
                     else:
                         raise
         return True
 
+    @api.v7
     def check(self, cr, uid, ids, autocommit=False, context=None):
+        return self.browse(cr, uid, ids, context).check(autocommit)
+
+    @api.v8
+    @api.multi
+    def check(self, autocommit=False):
         done_ids = []
-        for procurement in self.browse(cr, uid, ids, context=context):
+        for procurement in self:
             try:
-                result = self._check(cr, uid, procurement, context=context)
+                result = self._check(procurement)
                 if result:
                     done_ids.append(procurement.id)
                 if autocommit:
-                    cr.commit()
+                    self._cr.commit()
             except OperationalError:
                 if autocommit:
-                    cr.rollback()
+                    self._cr.rollback()
                     continue
                 else:
                     raise
         if done_ids:
-            self.write(cr, uid, done_ids, {'state': 'done'}, context=context)
+            for procurement in self:
+                procurement.write({'state': 'done'})
         return done_ids
 
-    #
     # Method to overwrite in different procurement modules
-    #
-    def _find_suitable_rule(self, cr, uid, procurement, context=None):
+    @api.model
+    def _find_suitable_rule(self, procurement):
```

Comment:
Here @api.model method is not appropriate. You can make a method for @api.v7 and @api.v8 .